### PR TITLE
Improve Accessibility Compliance on the Location Page 

### PR DESCRIPTION
**Pull request by NataliiaPodp**
## Checklist

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [x] I have checked my code and corrected any misspellings
-   [x] My changes take different screen sizes into account
-   [x] My changes are backward compatible
    -   [x] I have not changed any parameter names
    -   [x] I have not changed any translation keys
-   [x] My changes conform to the contributing guidelines (`CONTRIBUTING.md`)

image

### CODE REVIEW

File: layouts/partials/menus/secondary.html [hunk 1]
```diff
@@ -3,18 +3,18 @@
...
                 class="menu-item__link"
                 href="{{- $item.URL }}"
                 role="menuitem"
                 aria-label="{{ $title }}"
-                aria-details="{{- $item.Params.description | plainify }}">
+                aria-describedby="description-{{ $index }}">
                 <h1 class="heading heading--3 heading--embedded">{{- $title }}</h1>
                 <div class="secondary-menu__description">
-                    <p class="paragraph paragraph-embedded">
+                    <p class="paragraph paragraph-embedded" id="description-{{ $index }}">
```
Comment (lukket) [inline layouts/partials/menus/secondary.html:17]: I suggest choosing a more precise ID to avoid conflicts with other IDs. Something like `secondary-menu-description-XYZ`).

**Review by lukket — DISMISSED** [review]

File: layouts/partials/menus/secondary.html (hunk 1 above)
Comment (NataliiaPodp) [reply layouts/partials/menus/secondary.html]: done

File: i18n/en.yaml [hunk 2]
```diff
@@ -25,6 +25,8 @@ menu.location: Location
 menu.sponsors: Sponsors & Partners
 location_page.heading: Our Location
 location_page.directions_heading: Directions
+location_page.map.frame: Interactive map showing the location of the event
+location_page.map.fullscreen: View larger map of the event location
```
Comment (lukket) [inline i18n/en.yaml:29]: Please check the guidelines ☺️:

```suggestion
location_page.map_frame: Interactive map showing the location of the event
location_page.map_fullscreen: View larger map of the event location
```

File: i18n/de.yaml [hunk 3]
```diff
@@ -24,6 +24,8 @@ menu.location: Location
 menu.sponsors: Sponsoren & Partner
 location_page.heading: Unsere Location
 location_page.directions_heading: Anfahrt
+location_page.map.frame: Interaktive Karte, die den Standort der Veranstaltung zeigt
+location_page.map.fullscreen: Größere Karte des Veranstaltungsortes anzeigen
```
Comment (lukket) [inline i18n/de.yaml:28]: Same here:
```suggestion
location_page.map_frame: Interaktive Karte, die den Standort der Veranstaltung zeigt
location_page.map_fullscreen: Größere Karte des Veranstaltungsortes anzeigen
```

File: layouts/partials/map-embedding.html [hunk 4]
```diff
@@ -2,11 +2,20 @@
 
 
 <div class="map-container">
-    <iframe class="map-container__iframe" scrolling="no" allowfullscreen src="{{ .embeddableMapUrl | htmlUnescape }}"></iframe>
+    <iframe
+        class="map-container__iframe"
+        scrolling="no"
+        allowfullscreen
+        src="{{ .embeddableMapUrl | htmlUnescape }}"
+        title="{{ T "location_page.map.frame" }}"></iframe>
```
Comment (lukket) [inline layouts/partials/map-embedding.html:10]:
```suggestion
        title="{{ T "location_page.map_frame" }}"></iframe>
```

File: layouts/partials/map-embedding.html [hunk 5]
```diff
@@ -2,11 +2,20 @@
...
 
     {{- if .viewLargerMapUrl }}
-        <a href="{{ .viewLargerMapUrl }}" rel="external" class="map-container__button">
-            <img src="{{ $icon.RelPermalink | htmlUnescape }}" alt="Location Icon" width="18" height="18" />
+        <a
+            href="{{ .viewLargerMapUrl }}"
+            rel="external"
+            class="map-container__button"
+            aria-label="{{ T "location_page.map.fullscreen" }}">
```
Comment (lukket) [inline layouts/partials/map-embedding.html:17]:
```suggestion
            aria-label="{{ T "location_page.map_fullscreen" }}">
```

File: layouts/partials/menus/sidebar.html [hunk 6]
```diff
@@ -1,20 +1,22 @@
...
-                    class="sidebar-menu__link sidebar-menu__link--active"
-                {{- else }}
-                    class="sidebar-menu__link"
-                {{- end }}>
-                {{- or (T (printf "menu.%s" .Identifier)) .Name }}
-            </a>
-        </li>
-    {{- end }}
-</menu>
+<nav role="navigation">
```
Comment (lukket) [inline layouts/partials/menus/sidebar.html:4]: Why this change? Why no menu anymore? The `baseof` template already wraps the menu and everything else of the sidebar into a `nav` element.

Comment (NataliiaPodp) [reply layouts/partials/menus/sidebar.html:4]: I didn't check the usage. Thanks. The `nav` could be removed from here at all. But I would suggest using `ul` instead of `menu` - [docs](https://www.w3.org/WAI/tutorials/menus/structure/#unordered-list).

Comment (NataliiaPodp) [reply layouts/partials/menus/sidebar.html]: Here is the warning about the menu:
image

File: layouts/_default/baseof.html [hunk 7]
```diff
@@ -104,15 +104,13 @@
...
                 }}
 
+                {{- partial "menus/sidebar.html" (
+                    dict
+                    "menu" .Site.Menus.main
+                    "context" .
+                    )
+                }}
 
-                <div class="page-sidebar__menu">
```
Comment (lukket) [inline layouts/_default/baseof.html:108]: There is still the `page-sidebar__menu` style that we can remove then?

File: layouts/partials/menus/sidebar.html (hunk 6 above)
Comment (lukket) [reply layouts/partials/menus/sidebar.html]: Let me try to sum up my findings. If we use `<ul>` or `<menu>` makes absolutely no difference.

> The `<menu>` element is described in the HTML specification as a semantic alternative to `<ul>`,
> but treated by browsers (and exposed through the accessibility tree) as no different than `<ul>`.
> -- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/menu

So we can continue using `<menu>` here to clarify (under developers) that this is not just another unordered list. The actual problem here, that lead to the [error](https://dequeuniversity.com/rules/axe/4.6/aria-required-parent), was that we used `role="menuitem"` but forgot the `role="menu"` here on the unordered list. However, the roles `menu` and `menuitem` seem not necessary (but also not wrong?) when used inside a `<nav>` element.

Finally, what's written about the [`navigation` role on MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/navigation_role) is that each navigation should have a label (e.g., `Main`, `Secondary`, `Footer`). And since we have two navigation elements that are actually the same (header and sidebar), they should have the same label (`Main`). But also: "`<nav>` is intended only for a major block of navigation links; typically the `<footer>` element often has a list of links that don't need to be in a `<nav>` element." ([MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nav#attributes)). So I guess we should remove the `<nav>` element from the footer and secondary menu.

Comment (lukket) [reply layouts/partials/menus/sidebar.html]: To stay consistent, I would keep the `<nav>` elements as is. We just should label them. In our four menu partials, I would stay with `<menu>` for the given reasons. If you change it to `<ul>`, please change all of them.

Comment (NataliiaPodp) [reply layouts/partials/menus/sidebar.html]: This issue was discussed during our recent call. We agreed to resolve this issue by following [mdn web docs](https://developer.mozilla.org/en-US/) and [W3C WAI](https://www.w3.org/WAI/tutorials/menus/)

File: tests/featured-speaker.spec.ts [hunk 8]
```diff
@@ -24,10 +24,19 @@ import { expect, test } from '@playwright/test';
 ].forEach(({ fullName, tagLine, profileUrl }) => {
     test(`Should provide link to page of featured speaker ${fullName} on home page`, async ({ page }) => {
         await page.goto('/');
+
         const featuredSpeakersSection = page.getByRole('region', { name: 'Featured Speakers' });
         const featuredSpeakersMenu = featuredSpeakersSection.getByRole('menu', { name: 'Featured Speakers' });
         const featuredSpeakersMenuItem = featuredSpeakersMenu.getByRole('menuitem', { name: fullName });
-        await expect(featuredSpeakersMenuItem.getByText(fullName)).toBeVisible();
+
+        // The Playwright's strict mode throws an error when multiple elements
```
Comment (lukket) [inline tests/featured-speaker.spec.ts:32]: All this text seems a good reason to clean this up a little bit 😅. Not sure why I implemented it as `<span class="speaker-card__name">{{ $input.name }}</span>`, but from a user perspective, I would say it's just a header. `<h1 class="heading">{{ $input.name }}</h1>` results in the exact same layout. This expectation can easily be implemented as test (`featuredSpeakersMenu.getByRole('heading', { name: fullName })`, `featuredSpeakersMenu.getByRole('paragraph', { name: tagline })`, etc).

**Review by lukket — DISMISSED** [review]

File: tests/featured-speaker.spec.ts (hunk 8 above)
Comment (NataliiaPodp) [reply tests/featured-speaker.spec.ts]: done

File: tests/featured-speaker.spec.ts [hunk 9]
```diff
@@ -5,31 +5,39 @@ import { expect, test } from '@playwright/test';
...
         await page.goto('/');
+
         const featuredSpeakersSection = page.getByRole('region', { name: 'Featured Speakers' });
-        const featuredSpeakersMenu = featuredSpeakersSection.getByRole('menu', { name: 'Featured Speakers' });
-        const featuredSpeakersMenuItem = featuredSpeakersMenu.getByRole('menuitem', { name: fullName });
-        await expect(featuredSpeakersMenuItem.getByText(fullName)).toBeVisible();
-        await expect(featuredSpeakersMenuItem.getByText(tagLine)).toBeVisible();
-        await featuredSpeakersMenuItem.getByRole('link', { name: fullName }).click();
+        const featuredSpeakersList = featuredSpeakersSection.getByRole('list', { name: 'Featured Speakers' });
+        const featuredSpeakerItem = featuredSpeakersList.locator(`li[data-testid="speaker-item-${index}"]`);
```
Comment (lukket) [inline tests/featured-speaker.spec.ts:34]: Some thoughts on the `locator()`:

- Both, [Testing Library](https://testing-library.com/docs/queries/about#priority) and [Playwright](https://playwright.dev/docs/locators#quick-guide) (they took it over from Testing Library) recommend not to use CSS selectors (or only as last resort). The reason Testing Library exist is that CSS selectors create so many false-positives and false-negatives.
- There is a locator for test IDs (`getByTestId`), but still it's the last option in the list.
- We can still go by role here. From a semantic view, we can expect a list and list items here.

I'll update the test, and then we can close this PR. Great job 🫶🏻

**Review by lukket — DISMISSED** [review]
